File: algorithms/cpp/minimumDistancetotheTargetElement/minimumDistancetotheTargetElement.cpp
// Source: https://leetcode.com/problems/minimum-distance-to-the-target-element/
// Author: Miao Zhang
// Date:   2021-06-11

class Solution {
public:
    int getMinDistance(vector<int>& nums, int target, int start) {
        int res = INT_MAX;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == target)
                res = min(res, abs(i - start));
        }
        return res;
    }
};

File: algorithms/python/minimumDistancetotheTargetElement/minimumDistancetotheTargetElement.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/minimum-distance-to-the-target-element/
# Author: Miao Zhang
# Date:   2021-06-11

class Solution:
    def getMinDistance(self, nums: List[int], target: int, start: int) -> int:
        res = float('inf')
        for i in range(len(nums)):
            if nums[i] == target:
                res = min(res, abs(i - start))
        return res
